server.js

```diff
@@ -772,6 +772,11 @@ cache(function(data, match, sendBadge, request) {
       return;
     }
     try {
+      if (res.statusCode === 404) {
+        badgeData.text[1] = 'project not found or access denied';
+        sendBadge(format, badgeData);
+        return;
+      }
```

Comment:
Thanks for this improvement. Are you up for adding tests of the main appveyor badge while you're here, and since it's also affected by this change?

Reply:
I've added six tests. three for the existing build badge, and three for the new test badge

Reply:
I misread that. My bad!